tests/conftest.py

```diff
@@ -137,23 +130,135 @@ def pytest_generate_tests(metafunc):
                 assert o in K8S_SUPPORTED_OBSERVERS, "observer \"%s\" not supported!" % o
             observers_to_test = k8s_observers.split(',')
         metafunc.parametrize("k8s_observer", observers_to_test, ids=[o for o in observers_to_test], indirect=True)
-    k8s_monitors = metafunc.config.getoption("--k8s-monitors")
-    monitors_without_endpoints_to_test = []
-    monitors_with_endpoints_to_test = []
-    if not k8s_monitors:
-        monitors_without_endpoints_to_test = [m for m in MONITORS_WITHOUT_ENDPOINTS if m["type"] in K8S_DEFAULT_MONITORS_WITHOUT_ENDPOINTS]
-        monitors_with_endpoints_to_test = [m for m in MONITORS_WITH_ENDPOINTS if m[0]["type"] in K8S_DEFAULT_MONITORS_WITH_ENDPOINTS]
-    elif k8s_monitors.lower() == 'all':
-        monitors_without_endpoints_to_test = MONITORS_WITHOUT_ENDPOINTS
-        monitors_with_endpoints_to_test = MONITORS_WITH_ENDPOINTS
+
+
+@pytest.fixture(scope="session")
+def local_registry(request, worker_id):
+    def get_free_port():
+        s = socket.socket()
+        s.bind(('', 0))
+        return s.getsockname()[1]
+
+    def teardown():
+        if cont and worker_id == "master":
+            cont.remove(force=True, v=True)
+
+    def registry_is_ready():
+        if container_is_running(client, "registry"):
+            cont = client.containers.get("registry")
+            return has_log_message(cont.logs().decode('utf-8'), message="listening on [::]:%d" % port)
+        return False
+        
+    client = get_docker_client()
+    cont = None
+    #port = get_free_port()
+    port = 5000
+    request.addfinalizer(teardown)
+    if worker_id == "master" or worker_id == "gw0":
+        #try:
```

Comment:
Remove commented out logic unless it is needed.

Reply:
In general, I would say if you can make the dynamic port assignment work, that would be better than having the tests fail due to something else running on port 5000.